### PR TITLE
recommend RHEL/CentOS 8, update docker-aio

### LINKED ISSUE #7394: recommend CentOS 8
The guides recommend RHEL/CentOS 7.

Recommend 8, and make note of the additional subscription / repositories necessary for Rserve under 8.

### PR DESCRIPTION
**Pull request by donsizemore**
**What this PR does / why we need it**: The guides recommend RHEL/CentOS 7, but 8 has been out for a while now.

**Which issue(s) this PR closes**:

Closes #7394

**Special notes for your reviewer**: none

**Suggestions on how to test this**: verify commands

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**: no

**Is there a release notes update needed for this change?**: no

**Additional documentation**: none

### CODE REVIEW

File: conf/docker-aio/c8.dockerfile
Comment (pdurbin) [inline conf/docker-aio/c8.dockerfile:1]: Since we're creating this new c8.dockerfile, should we go ahead and delete the c7.dockerfile in this directory?

File: conf/docker-dcm/c8client.dockerfile
Comment (pdurbin) [inline conf/docker-dcm/c8client.dockerfile:1]: Should we delete c6client.dockerfile since we're adding c8client.dockerfile?

File: doc/sphinx-guides/source/installation/prerequisites.rst
Comment (pdurbin) [inline doc/sphinx-guides/source/installation/prerequisites.rst:17]: Just a comment that it's sad to see "the steps below were orginally[sic] written for el6" in here. I'm not sure if the steps need to be rewritten or not. People would probably complain if they don't work. Maybe we should rewrite this line.

File: doc/sphinx-guides/source/installation/prerequisites.rst
Comment (pdurbin) [inline doc/sphinx-guides/source/installation/prerequisites.rst:325]:
```suggestion
For RHEL/CentOS, the EPEL distribution is strongly recommended:
```

To avoid this error:

/Users/pdurbin/github/iqss/guides.dataverse.org/doc/sphinx-guides/source/installation/prerequisites.rst:331: WARNING: Literal block expected; none found.

It's the double :: that's causing trouble. Removing one fixes it. You could change the remaining colon to a period. Probably looks better.

File: doc/sphinx-guides/source/installation/prerequisites.rst
Comment (pdurbin) [inline doc/sphinx-guides/source/installation/prerequisites.rst:339]: I'm new to CodeReady-Builder. Is it an alternative to EPEL?

File: doc/sphinx-guides/source/installation/prerequisites.rst
Comment (pdurbin) [inline doc/sphinx-guides/source/installation/prerequisites.rst:343]: I'm new to PowerTools as well. I'm starting to worry that there are a lot of steps here. Is it possible to simplify it, at least in how it's described at the top as "You're going to enable EPEL and then install a package called R-core-devel." Or the extra things, PowerTools and CodeReady-Builder really necessary?

**Review by pdurbin — COMMENTED** [review]
Overall, this is looking really good. One thing to point out to @djbrooke is the even if we update the docs to recommend CentOS 8, actually switching Harvard Dataverse (and perhaps various test/QA/demo boxes) is another matter. We know CentOS 8 works well in our spin up scripts, thanks to dataverse-ansible and all our recent Jenkins work (thank you @donsizemore !).

I left various comments. I think @pameyer should take a look (and he's willing to, I asked him) because we're proposing changing to his docker-aio and docker-dcm stuff.

File: doc/sphinx-guides/source/installation/prerequisites.rst
Comment (donsizemore) [reply doc/sphinx-guides/source/installation/prerequisites.rst:343]: yes, the R portion became a bit of a choose-your-own adventure, but to my knowledge these are the necessary steps. I forget the additional dependencies provided by each repo. I can write a friendlier introductory paragraph if you'd like (but first let me comment out these steps in dataverse-ansible to verify that they're still needed).

File: tests/jenkins/groupvars.yml
Comment (donsizemore) [reply tests/jenkins/groupvars.yml:2]: I was having to submit a PR and wait on the review process each time someone requested a new feature and accompanying group_var in dataverse-ansible. Since this file only applied beneath the dataverse-ansible role, I moved it there. Let me know if this should be removed in a separate issue.

File: conf/docker-dcm/c8client.dockerfile
Comment (pameyer) [reply conf/docker-dcm/c8client.dockerfile:1]: Probably - seems like it's less confusing to have two copies.

File: conf/docker-dcm/dcmsrv.dockerfile
Comment (pameyer) [reply conf/docker-dcm/dcmsrv.dockerfile:2]: It might make more sense to leave use centos:7 here - the DCM server side stuff is tested on that OS, but I haven't tested with Cent8 yet.  I don't expect any issues, but that's different from knowing that it works.

File: conf/docker-dcm/rsalsrv.dockerfile
Comment (pameyer) [inline conf/docker-dcm/rsalsrv.dockerfile:1]: This might also be better left at centos:7 (server-side component hasn't been tested on cent8).

File: conf/docker-aio/c8.dockerfile
Comment (pameyer) [reply conf/docker-aio/c8.dockerfile:1]: It might be useful for testing on cent7 and cent8; but that's probably better handled through dataverse-ansible.

File: conf/docker-aio/readme.md
Comment (pameyer) [inline conf/docker-aio/readme.md:30]: updating `prep_it.bash` with the corresponding change is probably also helpful.

File: doc/sphinx-guides/source/installation/prerequisites.rst
Comment (donsizemore) [reply doc/sphinx-guides/source/installation/prerequisites.rst:343]: Yes, we still need these repos: `Depsolve Error occured: \n Problem: conflicting requests\n  - nothing provides openblas-devel needed by R-core-devel-4.0.3-1.el8.x86_64\n  - nothing provides texinfo-tex needed by R-core-devel-4.0.3-1.el8.x86_64`

File: conf/openshift/openshift.json
Comment (donsizemore) [reply conf/openshift/openshift.json]: removed in 28d38c3a1

**Comment by pdurbin** [comment]
@donsizemore I did some testing and made a pull request for this pull request. Please see https://github.com/OdumInstitute/dataverse/pull/58

File: conf/docker-aio/readme.md
Comment (pdurbin) [reply conf/docker-aio/readme.md:30]: Thanks, fixed in 7f90961.

File: conf/docker-aio/c8.dockerfile
Comment (pdurbin) [reply conf/docker-aio/c8.dockerfile:1]: I removed the old one. We can always pull it out of git history if we need it.

File: conf/docker-dcm/dcmsrv.dockerfile
Comment (pdurbin) [reply conf/docker-dcm/dcmsrv.dockerfile]: I reverted it back to CentOS 6. I don't want to retest DCM stuff.

File: tests/jenkins/groupvars.yml
Comment (pdurbin) [reply tests/jenkins/groupvars.yml:2]: No, it's fine. Thanks.

File: conf/docker-dcm/c8client.dockerfile
Comment (pdurbin) [reply conf/docker-dcm/c8client.dockerfile]: In the end we decided to stick with the older, already tested version of CentOS for DCM stuff.

File: conf/docker-dcm/rsalsrv.dockerfile
Comment (pdurbin) [reply conf/docker-dcm/rsalsrv.dockerfile]: Reverted. Thanks.

**Review by pdurbin — APPROVED** [review]
Looks good. Something to think about though: Are we ok recommending CentOS 8 in the guides when we use CentOS 7 in production?

**Comment by donsizemore** [comment]
> Looks good. Something to think about though: Are we ok recommending CentOS 8 in the guides when we use CentOS 7 in production?

I am. RHEL7 EOLs in June 2024; RHEL8 EOLs in 2029. CI testing already happens in CentOS 8 and I'm doing all new installs there. I would set forth that the Java version would make a much bigger difference than the OS version.

**Pull request merged**
